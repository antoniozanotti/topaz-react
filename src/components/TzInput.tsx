import React, { forwardRef } from 'react';
import { useVariantClasses, useFocusClasses } from './useStyles';

export interface TzInputProps extends React.ComponentProps<'input'> {
  variant?: 'accent' | 'primary' | 'secondary' | 'negative' | 'dark' | 'light';
  filled?: boolean;
}

export const TzInput = forwardRef<HTMLInputElement, TzInputProps>(
  function TzInput(
    { variant = 'accent', filled = true, ...props }: TzInputProps,
    ref
  ) {
    /* input classes */
    let inputOtherClasses = 'rounded';

    // focus
    let focusClasses = useFocusClasses();

    // size
    let sizeClasses =
      'h-[28px] sm:h-[38px] lg:h-[48px] px-[10px] sm:px-[14px] lg:px-[20px]';

    // variants and filled
    let variantClasses = useVariantClasses(variant, filled);

    let inputClasses = `${inputOtherClasses} ${focusClasses} ${sizeClasses} ${variantClasses}`;
    props.className = props.className
      ? inputClasses + ' ' + props.className
      : inputClasses;

    return <input {...props} ref={ref} />;
  }
);
