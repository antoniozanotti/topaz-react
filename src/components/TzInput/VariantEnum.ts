export enum VariantEnum {
  ACCENT = 'accent',
  PRIMARY = 'primary',
  SECONDARY = 'secondary',
  NEGATIVE = 'negative',
  DARK = 'dark',
  LIGHT = 'light',
}