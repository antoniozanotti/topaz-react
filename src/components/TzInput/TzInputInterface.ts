import { VariantEnum } from "./VariantEnum";

export interface TzInputInterface extends React.ComponentProps<'input'> {
  variant?: VariantEnum;
  filled?: boolean;
}