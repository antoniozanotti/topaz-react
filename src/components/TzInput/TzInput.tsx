import React, { forwardRef } from 'react';
import { TzInputInterface } from './TzInputInterface';
import { VariantEnum } from './VariantEnum';
import { useVariantStyles, useFocusStyles } from '../useStyles';

export const TzInput = forwardRef<HTMLInputElement, TzInputInterface>(
  function TzInput(
    { variant = VariantEnum.ACCENT, filled = true, ...props }: TzInputInterface,
    ref
  ) {
    let inputBaseStyles = 'rounded';
    let focusStyles = useFocusStyles();
    let sizeStyles =
      'h-[28px] sm:h-[38px] lg:h-[48px] px-[10px] sm:px-[14px] lg:px-[20px]';
    let variantStyles = useVariantStyles(variant, filled);
    let inputStyles = `${inputBaseStyles} ${focusStyles} ${sizeStyles} ${variantStyles}`;
    props.className = props.className
      ? `${inputStyles} ${props.className}`
      : inputStyles;

    return <input {...props} ref={ref} />;
  }
);
