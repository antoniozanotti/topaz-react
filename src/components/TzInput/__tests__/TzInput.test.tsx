import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { TzInput } from '../TzInput';

describe('TzInput Component', () => {
  it('should render correctly', () => {
    const { container } = render(<TzInput name="inputName" />);

    const input = container.querySelector("input[name='inputName']");

    expect(input).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
