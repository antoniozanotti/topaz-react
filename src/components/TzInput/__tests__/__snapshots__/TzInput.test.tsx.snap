// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TzInput Component should render correctly 1`] = `
<div>
  <input
    class="rounded focus:outline focus:outline-offset-2 focus:outline-2 h-[28px] sm:h-[38px] lg:h-[48px] px-[10px] sm:px-[14px] lg:px-[20px]  bg-accent dark:bg-dark-accent hover:bg-accent-1 active:bg-accent-2 focus:outline-accent dark:hover:bg-dark-accent-1 dark:active:bg-dark-accent-2 dark:focus:outline-dark-accent text-inverse-accent dark:text-inverse-dark-accent"
    name="inputName"
  />
</div>
`;
